osu.Game/Overlays/News/Sidebar/MonthPanel.cs

```diff
@@ -0,0 +1,152 @@
+
+                if (open.NewValue)
+                {
+                    AutoSizeAxes = Axes.Y;
+                    postsFlow.FadeIn(250, Easing.OutQuint);
+                }
+                else
+                {
+                    AutoSizeAxes = Axes.None;
+                    this.ResizeHeightTo(15, 250, Easing.OutQuint);
+
+                    postsFlow.FadeOut(250, Easing.OutQuint);
+                }
+            }, true);
+
+            // First state change should be instant.
+            FinishTransforms();
+            postsFlow.FinishTransforms();
+        }
+
+        private class DropdownButton : OsuHoverContainer
+        {
+            public readonly BindableBool IsOpen = new BindableBool();
+
+            protected override IEnumerable<Drawable> EffectTargets => null;
+
+            private readonly SpriteIcon icon;
+
+            public DropdownButton(DateTimeOffset date)
+            {
+                Size = new Vector2(160, 15);
+                Action = IsOpen.Toggle;
+                Children = new Drawable[]
+                {
+                    new OsuSpriteText
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                        Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                        Text = date.ToString("MMM yyyy")
+                    },
+                    icon = new SpriteIcon
+                    {
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Size = new Vector2(10),
+                        Icon = FontAwesome.Solid.ChevronDown
+                    }
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                IsOpen.BindValueChanged(open =>
+                {
+                    icon.Scale = new Vector2(1, open.NewValue ? -1 : 1);
+                }, true);
+            }
+        }
+
+        private class PostButton : OsuHoverContainer
+        {
+            protected override IEnumerable<Drawable> EffectTargets => new[] { text };
+
+            private readonly TextFlowContainer text;
+
+            public PostButton(APINewsPost post)
+            {
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+
+                Child = text = new TextFlowContainer(t => t.Font = OsuFont.GetFont(size: 12))
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Text = post.Title
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                IdleColour = colourProvider.Light2;
+                HoverColour = colourProvider.Light1;
+                Action = () => { }; // TODO
```

Comment:
as an explanation for this that wasn't given, this is being set because otherwise `ClickableContainer` will spawn in a disabled state, which is a problem because all of these components aren't even supposed to *be* in a disabled state. i guess this *will* go away, though, eventually.

Reply:
at least use `Enabled.Value = true` rather than empty TODO comment that just raises questions.